src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,774 @@
+                    unique_id=str(transaction[_TRANID]),
+                    raw_data=json.dumps(transaction),
+                    timestamp=self._rp2timestamp_from_ms_epoch(transaction[_DIVTIME]),
+                    asset=transaction[_ASSET],
+                    exchange=self.__BINANCE_COM,
+                    holder=self.account_holder,
+                    transaction_type=transaction_type.value,
+                    spot_price=Keyword.UNKNOWN.value,
+                    crypto_in=str(amount),
+                    crypto_fee=None,
+                    fiat_in_no_fee=None,
+                    fiat_in_with_fee=None,
+                    fiat_fee=None,
+                    notes=notes,
+                )
+            )
+
+    def _process_sell(self, transaction: Any, out_transaction_list: List[OutTransaction], notes: Optional[str] = None) -> None:
+        self.__logger.debug("Sell Transaction: %s", transaction)
+        trade = self._to_trade(transaction[_SYMBOL], str(transaction[_AMOUNT]), str(transaction[_COST]))
+
+        # For some reason CCXT outputs amounts in float
+        if transaction[_SIDE] == _BUY:
+            out_asset = trade.quote_asset
+            in_asset = trade.base_asset
+            crypto_out_no_fee: RP2Decimal = RP2Decimal(str(transaction[_COST]))
+            conversion_info = f"{trade.quote_info} -> {trade.base_info}"
+        elif transaction[_SIDE] == _SELL:
+            out_asset = trade.base_asset
+            in_asset = trade.quote_asset
+            crypto_out_no_fee = RP2Decimal(str(transaction[_AMOUNT]))
+            conversion_info = f"{trade.base_info} -> {trade.quote_info}"
+        else:
+            raise Exception(f"Internal error: unrecognized transaction side: {transaction[_SIDE]}")
+
+        if transaction[_FEE][_CURRENCY] == out_asset:
+            crypto_fee: RP2Decimal = RP2Decimal(str(transaction[_FEE][_COST]))
+        else:
+            crypto_fee = ZERO
+        crypto_out_with_fee: RP2Decimal = crypto_out_no_fee + crypto_fee
+
+        # Binance does not report the value of transaction in fiat
+        out_transaction_list.append(
+            OutTransaction(
+                plugin=self.__BINANCE_COM,
+                unique_id=transaction[_ID],
+                raw_data=json.dumps(transaction),
+                timestamp=self._rp2timestamp_from_ms_epoch(transaction[_TIMESTAMP]),
+                asset=out_asset,
+                exchange=self.__BINANCE_COM,
+                holder=self.account_holder,
+                transaction_type=Keyword.SELL.value,
+                spot_price=Keyword.UNKNOWN.value,
+                crypto_out_no_fee=str(crypto_out_no_fee),
+                crypto_fee=str(crypto_fee),
+                crypto_out_with_fee=str(crypto_out_with_fee),
+                fiat_out_no_fee=None,
+                fiat_fee=None,
+                notes=(
+                    f"{notes + '; ' if notes else ''} Sell side of conversion from "

```

Comment:
Is this a conversion or a plain sell?

Reply:
At the moment, the plugin can only handle conversions because we will need the rate conversion system to handle plain sells (I think). Binance.com handles around 6-7 fiat that aren't stable coins, currently. There are several fiat that were supported at one time, but have been removed, like JPY or USD. To have full compatibility we will need to have support for all, but I'd like to know how to record the iso code. 

Once the rate conversion feature is established, I can add extra processing to `_process_sell` and `_process_buy` to handle the details.

Reply:
Sounds good and pair conversion infra has been merged.